src/net_processing.cpp

```diff
@@ -5830,7 +5897,18 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
         if (CanServeBlocks(*peer) && ((sync_blocks_and_headers_from_peer && !IsLimitedPeer(*peer)) || !m_chainman.IsInitialBlockDownload()) && state.vBlocksInFlight.size() < MAX_BLOCKS_IN_TRANSIT_PER_PEER) {
             std::vector<const CBlockIndex*> vToDownload;
             NodeId staller = -1;
-            FindNextBlocksToDownload(*peer, MAX_BLOCKS_IN_TRANSIT_PER_PEER - state.vBlocksInFlight.size(), vToDownload, staller);
+            auto get_inflight_budget = [&state]() {
+                return std::max(0, MAX_BLOCKS_IN_TRANSIT_PER_PEER - static_cast<int>(state.vBlocksInFlight.size()));
+            };
+
+            FindNextBlocksToDownload(*peer, get_inflight_budget(), vToDownload, staller);
```

Comment:
a42510ba9656aada250c5fdf402e3914ba6f51db: maybe add a comment above here that we prioritize new blocks in the snapshot chainstate by adding them to `vToDownload` first.

Reply:
Added a comment.